server/src/main/java/org/opensearch/cluster/metadata/DecommissionAttributeMetadata.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.metadata;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.Version;
+import org.opensearch.cluster.AbstractNamedDiffable;
+import org.opensearch.cluster.NamedDiff;
+import org.opensearch.cluster.decommission.DecommissionAttribute;
+import org.opensearch.cluster.decommission.DecommissionStatus;
+import org.opensearch.cluster.metadata.Metadata.Custom;
+import org.opensearch.common.Nullable;
+import org.opensearch.common.Strings;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.Objects;
+
+/**
+ * Contains metadata about decommission attribute
+ *
+ * @opensearch.internal
+ */
+public class DecommissionAttributeMetadata extends AbstractNamedDiffable<Custom> implements Custom {
+
+    public static final String TYPE = "decommissionedAttribute";
+
+    private final DecommissionAttribute decommissionAttribute;
+    private final DecommissionStatus status;
+    public static final String attributeType = "awareness";
+
+    /**
+     * Constructs new decommission attribute metadata with given status
+     *
+     * @param decommissionAttribute attribute details
+     * @param status                current status of the attribute decommission
+     */
+    public DecommissionAttributeMetadata(DecommissionAttribute decommissionAttribute, DecommissionStatus status) {
+        this.decommissionAttribute = decommissionAttribute;
+        this.status = status;
+    }
+
+    /**
+     * Constructs new decommission attribute metadata with status as {@link DecommissionStatus#INIT}
+     *
+     * @param decommissionAttribute attribute details
+     */
+    public DecommissionAttributeMetadata(DecommissionAttribute decommissionAttribute) {
+        this.decommissionAttribute = decommissionAttribute;
```

Comment:
may be use `this(decommissionAttribute, DecommissionStatus.INIT)`?

Reply:
Sure. Updating in next commit